python/lsst/daf/butler/_butler.py

```diff
@@ -1198,6 +1202,19 @@ def pruneCollection(self, name: str, purge: bool = False, unstore: bool = False)
         if collectionType is not CollectionType.RUN and purge:
             raise PurgeUnsupportedPruneCollectionsError(collectionType)
 
+        def remove(child: str, parent: str) -> None:
+            """Remove a child collection from a parent collection."""
+            # Remove child from parent.
+            chain = list(self.registry.getCollectionChain(parent))
+            try:
+                chain.remove(name)
+            except ValueError as e:
+                raise RuntimeError(f"{name} is not a child of {parent}") from e
+            self.registry.setCollectionChain(parent, chain)
+
+        if (unlink):
+            for parent in unlink:
+                remove(name, parent)
```

Comment:
I think we should move this down into the transaction block so the whole thing stays exception-safe.